cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/WaterContractEndpointInput.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    public static final class GetAll extends EndpointInput {
+        private final String projectId;
+        private final String officeId;
+        private final String waterUser;
+
+        private GetAll(String officeId, String projectId, String waterUser) {
+            this.officeId = Objects.requireNonNull(officeId);
+            this.projectId = Objects.requireNonNull(projectId);
+            this.waterUser = Objects.requireNonNull(waterUser);
+        }
+
+        String projectId() {
+            return this.projectId;
+        }
+
+        String waterUser() {
+            return this.waterUser;
+        }
+
+        String officeId() {
+            return this.officeId;
+        }
+
+        @Override
+        protected HttpRequestBuilder addInputParameters(HttpRequestBuilder httpRequestBuilder) {
+            return httpRequestBuilder.addQueryHeader(ACCEPT_QUERY_HEADER, ACCEPT_HEADER_V1);
+        }
+    }
+
+    public static final class GetOne extends EndpointInput {
+        private final String contractName;
+        private final String officeId;
+        private final String projectId;
+        private final String waterUser;
+
+        private GetOne(String officeId, String projectId, String waterUser, String contractName) {
+            this.contractName = Objects.requireNonNull(contractName);
+            this.officeId = Objects.requireNonNull(officeId);
+            this.projectId = Objects.requireNonNull(projectId);
+            this.waterUser = Objects.requireNonNull(waterUser);
+        }
+
+        String contractName() {
+            return contractName;
+        }
+
+        String officeId() {
+            return officeId;
+        }
+
+        String projectId() {
+            return projectId;
+        }
+
+        String waterUser() {
+            return waterUser;
+        }
+
+        @Override
+        protected HttpRequestBuilder addInputParameters(HttpRequestBuilder httpRequestBuilder) {
+            return httpRequestBuilder.addQueryHeader(ACCEPT_QUERY_HEADER, ACCEPT_HEADER_V1);
+        }
+    }
+
+    public static final class Delete extends EndpointInput {
+        static final String METHOD_QUERY_PARAMETER = "method";
+        private final String contractName;
+        private final String officeId;
+        private DeleteMethod deleteMethod = DeleteMethod.ALL;
+        private final String projectId;
+        private final String waterUserId;
+
+        private Delete(String officeId, String projectId, String waterUser, String contractName) {
+            this.contractName = Objects.requireNonNull(contractName);
+            this.officeId = Objects.requireNonNull(officeId);
+            this.projectId = Objects.requireNonNull(projectId);
+            this.waterUserId = Objects.requireNonNull(waterUser);
+        }
+
+        public Delete deleteMethod(DeleteMethod deleteMethod) {
+            this.deleteMethod = Objects.requireNonNull(deleteMethod);
```

Comment:
Remove the require non null, this is an optional field and null is allowed

Reply:
Removed non-null requirement